servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/QueueDynamicCompositeCancellable.java

```diff
@@ -16,63 +16,67 @@
 package io.servicetalk.concurrent.api;
 
 import io.servicetalk.concurrent.Cancellable;
+import io.servicetalk.concurrent.internal.FlowControlUtils;
 
 import java.util.Queue;
 import java.util.concurrent.ConcurrentLinkedQueue;
-import java.util.concurrent.atomic.AtomicIntegerFieldUpdater;
+import java.util.concurrent.atomic.AtomicLongFieldUpdater;
 
+import static io.servicetalk.concurrent.internal.FlowControlUtils.tryIncrementIfNotNegative;
 import static io.servicetalk.concurrent.internal.ThrowableUtils.catchUnexpected;
 import static io.servicetalk.utils.internal.PlatformDependent.throwException;
+import static java.util.concurrent.atomic.AtomicLongFieldUpdater.newUpdater;
 
 final class QueueDynamicCompositeCancellable implements DynamicCompositeCancellable {
-    private static final AtomicIntegerFieldUpdater<QueueDynamicCompositeCancellable> cancelledUpdater =
-            AtomicIntegerFieldUpdater.newUpdater(QueueDynamicCompositeCancellable.class, "cancelled");
+    private static final AtomicLongFieldUpdater<QueueDynamicCompositeCancellable> sizeUpdater =
+            newUpdater(QueueDynamicCompositeCancellable.class, "size");
     @SuppressWarnings("unused")
-    private volatile int cancelled;
-
+    private volatile long size;
```

Comment:
Queue size is `int` users won't be able to add more elements, why do we need `long` here?

Reply:
`size()` is int, however the implementations which are not `array` backed could conceivably hold more than `Integer.MAX_VALUE` elements. These implementations just cap the value of `size()` to `Integer.MAX_VALUE`. I refactored so instead of tracking size I am now just using an atomic operation on the collection.

Reply:
🤯 looked at `ConcurrentLinkedQueue` impl, TIL